pages/food-groups/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useRef } from 'react'
+import FoodGroups from 'components/FoodGroups'
+
+import Auth from '@components/Auth'
+import { Button, Modal } from '@components/Custom'
+import { ACHIEVEMENT, useFirebase } from '@components/FirebaseContext'
+import { DRAGGING_STATE } from '@components/FoodGroups/Draggable'
+import {
+  ORIGIN_VECTOR2,
+  Vector2
+} from '@components/FoodGroups/Draggable/boundingbox'
+import {
+  CHARACTER_POSITIONS,
+  CharacterSpawner,
+  generateCharacterSet
+} from '@components/FoodGroups/Draggable/characterspawner'
+import {
+  FoodGroupCharacterImage,
+  FoodGroupCharacterImageDynamic
+} from '@components/FoodGroups/Draggable/types'
+import { GROUPS } from '@components/FoodGroups/groups'
+import {
+  draggableZoneStyle,
+  startZoneStyle
+} from '@components/FoodGroups/styles'
+import { State_ } from '@components/FoodGroups/types'
+import {
+  getCharacterData,
+  getFormatData
+} from '@components/NotionAPI/getCharacterData'
+
+const N_DRAGGABLE = 5
+const newArray = (v: any) => Array(N_DRAGGABLE).fill(v)
+
+interface Props {
+  notion_character_data: FoodGroupCharacterImage[]
+}
+
+const FoodGroupsPage: React.FC<Props> = ({ notion_character_data }: Props) => {
+  const { achievements, updateAchievementsDocument, user } = useFirebase()
```

Comment:
There are some major changes made to `FirebaseContext`. Please update the codes that are using the values and functions from this context.

Reply:
fixed